detection_rules/devtools.py

```diff
@@ -249,7 +254,7 @@ def prune_staging_area(target_stack_version: str, dry_run: bool, exception_list:
     }
     exceptions.update(exception_list.split(","))
 
-    target_stack_version = Version(target_stack_version)[:2]
+    target_stack_version = Version.parse(target_stack_version, optional_minor_and_patch=True)
```

Comment:
these ignored patch - how does this change handle that?

Reply:
Great question as this command is important to the backporting workflow.

The `optional_minor_and_patch` flag is only to give it a default patch of 0. `target_stack_version` from the backport workflow command relies on dynamically generated `target-branches.yml` file which ultimately loads from `stack-schema-map.yaml` where the supported stack versions are full semantic versioned. Along the way, these versions are splices to remove patch, but ultimately the comparison will be `full semantic version` < `full semantic version` where patch will always be 0, unless we manually run this command and provide a different `target-stack-version` value. So it should be fine for backporting purposes.